### PR TITLE
Fix/2986  fix translation and add tests

### PR DESCRIPTION
**Pull request by raphj**
#2986

## Checkpoints

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: functionnal_tests/cypress/integration/app_file/translation.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { PAGES } from '../../support/urls_commands.js'
+
+describe('App File', () => {
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      cy.visit(`/ui/workspaces/${workspace.workspace_id}/contents?type=file`)
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_file/translation.js:9]: There is not this URL in the PAGES variable? If not, do you see an obstacle to adding it? (If you decide not to add it, there is an `import` not used in this file.)

File: functionnal_tests/cypress/integration/account/translation.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { PAGES } from '../../support/urls_commands.js'
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/account/translation.js:1]: Unused import

File: functionnal_tests/cypress/integration/admin/translation.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { PAGES } from '../../support/urls_commands.js'
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/admin/translation.js:1]: unused import

File: functionnal_tests/cypress/integration/app_file/translation.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    })
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_file/translation.js:13]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/app_folder/translation.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { PAGES } from '../../support/urls_commands.js'
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_folder/translation.js:1]: unused import

File: functionnal_tests/cypress/integration/app_folder/translation.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      cy.visit(`/ui/workspaces/${workspace.workspace_id}/contents?type=file`)
+    })
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_folder/translation.js:13]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/app_gallery/translation.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+describe('App Gallery', function () {
+  before(function () {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.visit('/ui/workspaces/1/gallery')
+  })
+
+  it(`should have translations`, () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_gallery/translation.js:9]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/app_html-document/translation.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { PAGES } from '../../support/urls_commands.js'
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_html-document/translation.js:1]: unused import

File: functionnal_tests/cypress/integration/app_html-document/translation.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      cy.visit(`/ui/workspaces/${workspace.workspace_id}/contents?type=html-document`)
+    })
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_html-document/translation.js:13]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/app_workspace_advanced/translation.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+describe('App Workspace Advanced', function () {
+  const newDescription = 'description'
+  let workspaceId = 1
+  let workspaceDescription = ''
+
+  before(() => {
+      cy.resetDB()
+      cy.setupBaseDB()
+      cy.loginAs('administrators')
+      cy.visit("/ui/admin/workspace")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_workspace_advanced/translation.js:10]:
```suggestion
      cy.visit('/ui/admin/workspace')
```

File: functionnal_tests/cypress/integration/dashboard/translation.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      cy.visitPage({ pageName: p.DASHBOARD, params: { workspaceId: workspace.workspace_id } })
+    })
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/dashboard/translation.js:13]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/dashboard/translation.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    cy.get('[data-cy="contentTypeBtn_contents/thread"]').contains('Start a topic')
+
+    cy.changeLanguage('fr')
+    cy.get('[data-cy="contentTypeBtn_contents/thread"]').contains("Lancer une discussion")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/dashboard/translation.js:17]:
```suggestion
    cy.get('[data-cy="contentTypeBtn_contents/thread"]').contains('Lancer une discussion')
```

File: functionnal_tests/cypress/integration/home_page/translation.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+describe('content :: home_page', function () {
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.visit('/')
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/home_page/translation.js:9]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/home_page/translation.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+    cy.get('.homepagecard__title').contains('Welcome to Tracim')
+
+    cy.changeLanguage('fr')
+    cy.get('.homepagecard__title').contains("Bienvenue sur Tracim")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/home_page/translation.js:13]:
```suggestion
    cy.get('.homepagecard__title').contains('Bienvenue sur Tracim')
```

File: functionnal_tests/cypress/integration/login/translation.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
+describe('Login page', function () {
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.visit('/ui/login')
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/login/translation.js:8]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/login/translation.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+    cy.get('.loginpage__card__form__btnsubmit').contains('Connection')
+
+    cy.changeLanguage('fr')
+    cy.get('.loginpage__card__form__btnsubmit').contains("Connexion")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/login/translation.js:12]:
```suggestion
    cy.get('.loginpage__card__form__btnsubmit').contains('Connexion')
```

File: functionnal_tests/cypress/integration/search/translation.js [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+    cy.get(searchInput).type('This long sentence will probably never yield any result but I hope it will make the reviewer smile a bit.')
+    cy.get(searchButton).click()
+    cy.url().should('include', searchURL)
+  })
+
+  it("should have translations", () => {
+    cy.get('.searchResult__title').contains('Search results')
+
+    cy.changeLanguage('fr')
+    cy.get('.searchResult__title').contains("Résultats de la recherche")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/search/translation.js:23]:
```suggestion
    cy.get('.searchResult__title').contains('Résultats de la recherche')
```

File: functionnal_tests/cypress/integration/search/translation.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.visitPage({pageName: PAGES.HOME})
+    cy.get(searchInput).type('This long sentence will probably never yield any result but I hope it will make the reviewer smile a bit.')
+    cy.get(searchButton).click()
+    cy.url().should('include', searchURL)
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/search/translation.js:19]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/sidebar/translation.js [hunk 19]
```diff
@@ -0,0 +1,19 @@
+describe('content :: home_page', function () {
+  before(() => {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.visit('/ui')
+    cy.get('.sidebar__content__navigation__workspace__item__number').click()
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/sidebar/translation.js:10]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/integration/timeline/translation.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+    cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      workspaceId = workspace.workspace_id
+      cy.visitPage({pageName: PAGES.CONTENTS, params: {workspaceId: workspaceId}})
+      cy.createHtmlDocument(htmlDocTitle, workspaceId)
+    })
+    cy.get(contentHtmlDocGetter).click()
+    cy.get(cancelDocBtn).click()
+  })
+
+  it("should have translations", () => {
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/timeline/translation.js:25]:
```suggestion
  it('should have translations', () => {
```

File: functionnal_tests/cypress/support/commands.js [hunk 21]
```diff
@@ -164,3 +164,12 @@ Cypress.Commands.add('cleanSessionCookies', () => {
 Cypress.Commands.add('cancelXHR', () => {
   cy.visit('/api/v2/doc/')
 })
+
+Cypress.Commands.add('changeLanguage', (langCode) => {
+    cy.get("#headerDropdownMenuButton")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/support/commands.js:169]:
```suggestion
    cy.get('#headerDropdownMenuButton')
```

File: functionnal_tests/cypress/integration/app_file/translation.js (hunk 1 above)
Comment (gbonaspetti) [reply functionnal_tests/cypress/integration/app_file/translation.js:9]: This applies to all similar cases.

File: functionnal_tests/cypress/integration/app_agenda/translation.js [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+
+    cy.fixture('baseWorkspace').as('workspace').then(workspace =>
+      cy.visitPage({ pageName: PAGES.AGENDA, params: { workspaceId: workspace.workspace_id } })
+    )
+
+    cy.changeLanguage('en')
+    cy.get('.agendaPage__title').contains('Agenda of shared space My Workspace')
+
+    cy.changeLanguage('fr')
+    cy.get('.agendaPage__title').contains("Agenda de l'espace partagé My Workspace")
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_agenda/translation.js:29]:
```suggestion
    cy.get('.agendaPage__title').contains('Agenda de l'espace partagé My Workspace')
```

File: functionnal_tests/cypress/integration/admin/content_admin_user.js [hunk 23]
```diff
@@ -1,13 +1,12 @@
 import { PAGES } from '../../support/urls_commands'
 
-
 describe("An admin seeing a user's profile", () => {
```
[the remark this answers is not in the document]
Comment (raphj) [reply functionnal_tests/cypress/integration/admin/content_admin_user.js:3]: This one is ok, there is a quote in the string :-)

File: functionnal_tests/cypress/integration/app_agenda/translation.js (hunk 22 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/app_agenda/translation.js:29]: This one is ok too

File: functionnal_tests/cypress/integration/account/translation.js (hunk 2 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/account/translation.js:1]: It is, now. I used visitPage whenever possible. Thanks!

File: functionnal_tests/cypress/integration/admin/translation.js (hunk 3 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/admin/translation.js:1]: It is now.

File: functionnal_tests/cypress/integration/app_folder/translation.js (hunk 5 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/app_folder/translation.js:1]: done

File: functionnal_tests/cypress/integration/app_html-document/translation.js (hunk 8 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/app_html-document/translation.js:1]: done

File: functionnal_tests/cypress/integration/app_folder/translation.js [hunk 24]
```diff
@@ -6,17 +6,17 @@ describe('App Folder Advanced', function () {
...
     cy.loginAs('users')
     cy.fixture('baseWorkspace').as('workspace').then(workspace => {
-      cy.visit(`/ui/workspaces/${workspace.workspace_id}/contents?type=file`)
+      cy.visitPage({ pageName: PAGES.CONTENTS, params: { workspaceId: workspace.workspace_id } })
     })
   })
 
   it('should have translations', () => {
-    cy.get('.folder__header__name').contains('Received files')
+    cy.get('.workspace__header__title').contains('List of contents')
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_folder/translation.js:14]: Is that right? It doesn't look like App Folder Advanced to me.

Comment (raphj) [reply functionnal_tests/cypress/integration/app_folder/translation.js]: fixed this :-)

File: functionnal_tests/cypress/integration/app_thread/translation.js [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+      cy.visitPage({
+        pageName: PAGES.CONTENTS,
+        getters: { type: 'thread' },
+        params: { workspaceId: workspace.workspace_id }
+      })
+    })
+  })
+
+  it('should have translations', () => {
+    cy.get('.workspace__header__title').contains('List of threads')
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/app_thread/translation.js:18]: This is testing the workspace, not the app thread. Use the same logic you did in `functionnal_tests/cypress/integration/app_file/translation.js` .

File: functionnal_tests/cypress/integration/search/translation.js [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+const searchInput = '[data-cy=search__text]'
+const searchButton = '[data-cy=search__btn]'
+
+describe('Search', () => {
+  before(function () {
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.loginAs('users')
+    cy.visitPage({ pageName: PAGES.HOME })
+    cy.get(searchInput).type('This long sentence will probably never yield any result but I hope it will make the reviewer smile a bit.')
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/search/translation.js:14]: :smile:

File: pushpin_cypress_config/routes [hunk 27]
```diff
@@ -1 +1 @@
-* localhost:1337
+* localhost:6543
```
Comment (gbonaspetti) [inline pushpin_cypress_config/routes:1]: reset this, please

File: functionnal_tests/cypress/integration/app_thread/translation.js (hunk 25 above)
Comment (raphj) [reply functionnal_tests/cypress/integration/app_thread/translation.js]: fixed, I overlooked this one, thanks

**Pull request merged**